### ISSUE TITLE
Libspotify.release_name can sometimes be nil

### ISSUE THREAD

**Issue by Burgestrand** [Burgestrand/libspotify#6]
See Burgestrand/spotify#27. Sometimes `host_cpu` does not contain the version, for example on some release of raspberry pi (raspbian?), `host_cpu` will only return `arm`, and not `armv5hf` or similar.

**Comment by Burgestrand** [comment]
I am preparing for a new release to address this issue.

At the same time, I'm thinking of upgrading libspotify to make a release that will target hard float OS. It's possible that hard float is more stable for FFI.

**Comment by Burgestrand** [comment]
Preparational work in e54855ff72390e51850ff5b0aa6faddf57fb1a37.

**Comment by Burgestrand** [comment]
Added 22c0fb7607740d9c917f01d321d6cb4aa205b84e and 383112e253cac51ed384d9067442bd6bf57f5244.

New approach is a bit more complex, but probably easier to support the various naming schemes one must resort to. I've tried it on Mac OS, Linux 32bit, Linux 64bit, Raspberry PI (hard float), and Windows 32bit.

I have not tried it on raspberry pi soft float. Worst case they'll have to install hard float, or do manual libspotify installation.
